reference-architecture/azure-ansible/bastion.sh

```diff
@@ -0,0 +1,337 @@
+chmod 600 /root/.ssh/id_rsa
+
+sleep 30
+cat <<EOF > /root/setup_ssmtp.sh
+# \$1 = Gmail Account (Leave off @gmail.com ie user)
+# \$2 = Gmail Password
+# \$3 = Notification email address
+# Setup ssmtp mta agent for use with gmail
+yum -y install wget
+wget -c https://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-8.noarch.rpm
+rpm -ivh epel-release-7-8.noarch.rpm
+yum -y install ssmtp
+alternatives --set mta  /usr/sbin/sendmail.ssmtp
+systemctl stop postfix
+systemctl disable postfix
+cat <<EOFZ > /etc/ssmtp/ssmtp.conf
+root=postmaster
+mailhub=mail
+TLS_CA_File=/etc/pki/tls/certs/ca-bundle.crt
+mailhub=smtp.gmail.com:587   # SMTP server for Gmail
+Hostname=localhost
+UseTLS=YES
+UseSTARTTLS=Yes
+FromLineOverride=YES #TO CHANGE FROM EMAIL
+Root=\${3} # Redirect root email
+AuthUser=\${1}@gmail.com
+AuthPass=\${2}
+AuthMethod=LOGIN
+RewriteDomain=gmail.com
+EOFZ
+cat <<EOFZ > /etc/ssmtp/revaliases
+root:\${1}@gmail.com:smtp.gmail.com:587
+EOFZ
+EOF
+chmod +x /root/setup_ssmtp.sh
+# Ignore Error If It Dont work
+/root/setup_ssmtp.sh ${AUSERNAME} ${PASSWORD} ${RHNUSERNAME} || true
+
+echo "${RESOURCEGROUP} Bastion Host is starting software update" | mail -s "${RESOURCEGROUP} Bastion Software Install" ${RHNUSERNAME} || true
+# Continue Setting Up Bastion
+subscription-manager unregister 
+yum -y remove RHEL7
+rm -f /etc/yum.repos.d/rh-cloud.repo
+# Found that wildcard disable not working all the time - make sure
+yum-config-manager --disable epel
+yum-config-manager --disable epel-testing
+subscription-manager register --username $RHNUSERNAME --password $RHNPASSWORD
+subscription-manager attach --pool=$RHNPOOLID
+subscription-manager repos --disable="*"
+subscription-manager repos     --enable="rhel-7-server-rpms"     --enable="rhel-7-server-extras-rpms"
+subscription-manager repos     --enable="rhel-7-server-ose-3.3-rpms"
+yum -y install atomic-openshift-utils
+yum -y install git net-tools bind-utils iptables-services bridge-utils bash-completion httpd-tools
+yum -y install docker
+touch /root/.updateok
+sed -i -e "s#^OPTIONS='--selinux-enabled'#OPTIONS='--selinux-enabled --insecure-registry 172.30.0.0/16'#" /etc/sysconfig/docker
+                                                                                         
+cat <<EOF > /etc/sysconfig/docker-storage-setup
+DEVS=/dev/sdc
+VG=docker-vg
+EOF
+
+docker-storage-setup                                                                                                                                    
+systemctl enable docker
+systemctl start docker
+
+
+cat <<EOF > /etc/ansible/hosts
+[OSEv3:children]
+masters
+etcd
+nodes
+misc
+
+[OSEv3:vars]
+azure_resource_group=${RESOURCEGROUP}
+rhn_pool_id=${RHNPOOLID}
+openshift_install_examples=true
+deployment_type=openshift-enterprise
+openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]
```

Comment:
Actually plan is HTPASSWD is "emergency login", Ive done the work for keycloak already, and will be the auth solution. This has been the strategy since day one. With keycloak everything can be integrated. I've done initial testing with it, Im only missing the auto setup of the providers.